gensim/models/keyedvectors.py

```diff
@@ -1130,16 +1275,26 @@ def evaluate_word_analogies(self, analogies, restrict_vocab=300000, case_insensi
                 self.vocab = ok_vocab
                 ignore = {a, b, c}  # input words to be ignored
                 predicted = None
-                # find the most likely prediction using 3CosAdd (vector offset) method
-                # TODO: implement 3CosMul and set-based methods for solving analogies
-                sims = self.most_similar(positive=[b, c], negative=[a], topn=5, restrict_vocab=restrict_vocab)
+                # find the most likely prediction using 3CosAdd (vector offset) method or 3CosMul method
+
+                if method == '3CosAdd':
+                    sims = self.most_similar(positive=[b, c], negative=[a], topn=topk, restrict_vocab=restrict_vocab)
```

Comment:
This doesn't look backward compatible. Whereas before we got **top 5** predictions, now we only get **top 1** prediction.

And if the top 1 prediction happens to be ingored (`ignore` etc), we fail, whereas before we continued on to the first acceptable word.

This seems fundamentally broken, how does this pass our unit tests? Maybe the unit tests need expanding, too.

Reply:
I changed topk to be equal to 5 by default. Now it should be compartilble. 

"The(TooMany)ProblemsofAnalogicalReasoningwithWordVectors, A.Rogers, A.Drozd et al."
"Methodical Evaluation of Arabic Word Embeddings, Mohammed Elrazzaz  et al."
In the papers mentioned above, it is written that e.g. the top-3 means that for a successful prediction the expected one should among the top-3 similar words. 
At your case top-3 and considering your implementation it means that if the first one is a, b or c the it checks the second etc..

Reply:
If you want to change the break part of the code as it used to be and keep only the implementation of 3CosMul and 3CosAvg methods just tell me.

Reply:
The default logic for evaluation should stay the same.

Adding an option to "check more than one hit (after ignoring trivial hits of `a`, `b`, `c`)" is welcome, but must not change the existing logic, and must default to the previous default.

Reply:
Fixed.. The logic stays the same.